plugin/src/main/java/io/jenkins/plugins/analysis/core/model/ReportScanningTool.java

```diff
@@ -294,6 +294,9 @@ public FormValidation doCheckPattern(@AncestorInPath final AbstractProject<?, ?>
             if (!JENKINS.hasPermission(Item.CONFIGURE, project)) {
                 return FormValidation.ok();
             }
+            if ((pattern == null || pattern.trim().isEmpty()) && !canScanConsoleLog()) {
```

Comment:
This will break existing parsers that provide a default pattern like CheckStyle.

It would make sense to provide that improved check in a method like `hasPattern` (or similar).

Reply:
Apologies - I'm being a bit myopic and focussed purely on the Groovy use case...
How about:
```
            final boolean thereIsNoPatternConfigured = pattern == null || pattern.trim().isEmpty();
            final boolean thereIsNoDefaultPattern = getPattern().trim().isEmpty();
            if (thereIsNoPatternConfigured && thereIsNoDefaultPattern && !canScanConsoleLog()) {
                return FormValidation.error(Messages.ReportScanningTool_PatternIsEmptyAndConsoleParsingDisabled());
            }
```

Reply:
I think you need to write a method (or variable):
```
boolean hasPattern() {
    return StringUtils.isNotEmpty(getPattern()) || StringUtils.isNotEmpty(getDescriptor().getPattern());
}
```

Reply:
Done - I added a "hasDefaultPattern()" method and used that.